### PR TITLE
Document campaigns

### PR DESCRIPTION
**Pull request by Adamik10**
#### Description

This PR introduces documentation on campaigns - explaining the concept, usage and (in my opinion most importantly) how to get to show them in local development environmet. 

#### Checklist
- [x] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [x] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [x] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions
n/a

### CODE REVIEW

File: docs/campaigns.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+criteria for each campaign they create. The visibility criteria is based on
+search filter facets. Does that sound familiar? Yes, we use anoter API to get
+that very data in THIS project - in the search result app.
+The facets differ based on the search string the user uses for their search.
+
+As an example, the dpl-cms admin could wish to show a Harry Potter related
+campaign to all the users whose search string retreives search facets which
+have "Harry Potter" as one of the most relevant subjects.
+Campaigns in dpl-cms can use triggers such as subject, main language, etc.
+
```
Comment (spaceo) [inline docs/campaigns.md:28]: Would it be an idea to show an example of the hook being used, or are you afraid of that the documentation gets deprecated when/if the code changes?
This is an ancient problem that I do not have the golden answer to.

File: docs/campaigns.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+## How campaign setup works in dpl-cms
+
+Dpl-cms is a cms system based on Drupal, where the system administrators can set
+up campaigns they want to show to their users. Drupal also allows the cms system
+to act as an API endpoint that we then can contact from our apps.
+The cms administratots can specify the content (image, text) and the visibility
+criteria for each campaign they create. The visibility criteria is based on
+search filter facets. Does that sound familiar? Yes, we use anoter API to get
+that very data in THIS project - in the search result app.
+The facets differ based on the search string the user uses for their search.
```
Comment (spaceo) [inline docs/campaigns.md:22]:
```suggestion
Dpl-cms is a cms system based on Drupal, where the system administrators can set
up campaigns they want to show to their users. Drupal also allows the cms system
to act as an API endpoint that we then can contact from our apps.

The cms administrators can specify the content (image, text) and the visibility
criteria for each campaign they create. The visibility criteria is based on
search filter facets.
Does that sound familiar? Yes, we use another API to get that very data
in THIS project - in the search result app.
The facets differ based on the search string the user uses for their search.
```

**Review by spaceo — CHANGES_REQUESTED** [review]
Some typos needs fixing. Also consider if there should be a code example in  the documentation. I could imagine that it is a bit abstract for a person that does not know about the systems.

File: docs/campaigns.md (hunk 1 above)
Comment (Adamik10) [reply docs/campaigns.md:30]: You know what, that's not a bad idea :) I included a snippet 😉

**Review by spaceo — APPROVED** [review]
LGTM

**Review by jeppekroghitk — APPROVED** [review]

**Pull request merged**